app/Main.hs

```diff
@@ -199,9 +199,28 @@ updateImpl config@PackageConfig{ depends } = do
   echoT ("Updating " <> pack (show (length trans)) <> " packages...")
   forConcurrently_ trans . uncurry $ installOrUpdate (set config)
 
+pursCmd :: IO Text
+pursCmd = do
+  purs <- which "purs"
+  cmd <- which "purs.cmd"
+  exe <- which "purs.exe"
+  return $ fromMaybe "purs" $ msum [ constCmd "purs" purs
+                                   , findExe cmd
+                                   , constCmd "purs" exe
+                                   ]
+    where
+      constCmd :: Text -> Maybe Turtle.FilePath -> Maybe Text
+      constCmd t = fmap (const t)
+      appendExeText :: Turtle.FilePath -> Text
+      appendExeText p =
+        either id id (toText (p </> "node_modules" </> "purescript" </> "vendor" </> "purs.exe"))
```

Comment:
I'm not keen on depending on npm here like this. I would encourage users to put the executable on their PATH instead.

Reply:
Yeah... I can't argue with you there. 

The problem that I was working around was that when a user installs purescript with npm, the `purs.cmd` is what ends up on the path. That is executable, but when run as a proc, the path macros in the file don't expand properly. So `purs.cmd` has to be run with shell instead.

I did this because I was trying to avoid having to shell out for that one case, but this is kind of a terrible hack :man_shrugging: 

So, I could take a crack at shelling out when we find the `purs.cmd` but no exe on the path.

Or we could pop an error message when we detect this case that says "Windows users need to have the exe on their path". Maybe even include a link to the latest release or something.

Thoughts?